### ISSUE TITLE
ffmpeg + SDL

### ISSUE THREAD

**Issue by vcunat** [NixOS/nixpkgs#24658]
ffmpeg-3.2 (I think) [dropped support](http://ffmpeg.org/index.html#sdl1) for version 1 of SDL.  As a result, some stuff gets missing unexpectedly in our builds, e.g. ffplay in `ffmpeg-full` (which was just noticed on IRC).

Both 17.03 and unstable/master are affected.

**Comment by MP2E** [comment]
Partially fixed with PR #24698 

the regular ffmpeg package will be a little bit trickier due to generic.nix

**Comment by codyopel** [comment]
This is kinda ugly but is all that needs to be changed for SDL.
```nix
{ sdlSupport ? !stdenv.isArm, SDL, SDL2 }:
{
  buildInputs = optional sdlSupport (if minVer "3.2" then SDL2 else SDL);
  configureFlags = [
    (ifMinVer "2.5" (if ! minVer "3.2" && sdlSupport then "--enable-sdl" else null))
    (ifMinVer "3.2" (if sdlSupport then "--enable-sdl2" else null))
  ];
}
```
I partially scrapped this build and rewrote it a while back, if you look back in the commit history before we dropped support for older versions it should give you some ideas for a cleaner generic build.  This build also merges ffmpeg and ffmpeg-full into a single derivation.
https://github.com/triton/triton/tree/master/pkgs/all-pkgs/f/ffmpeg as well as the overrides in all-packages.nix

**Comment by MP2E** [comment]
My apologies, got side-tracked with work. Thanks codyopel, this is very helpful! I have created a PR based off of your comment. I think that in the long run, unifying ffmpeg and ffmpeg-full is a good idea. For now though, my PR should fully address this issue

**Comment by MP2E** [comment]
Fixed as of dfd371de9b011e319be6ee1fb53ed0dfcfe10431
